File: tree/gim_types.go
package tree

type TNode struct {
    Parent *TNode
    Children []*TNode
    Data string
    Type string
    Path string
}

func NewTNode() *TNode {
    return &TNode{nil, nil, "", "", ""}
}

type GimMasterTree struct {
    MasterDir string
    MasterRoot *TNode
    MasterRecordDir string
}

func NewGimMasterTree() GimMasterTree {
    return GimMasterTree{"", nil, ""}
}

type Blob struct {
    ShaHash string
    ModTime int64
    ActualPath string
}

File: tree_c/gim_types.c
#include<stdio.h>
#include <stdlib.h>
# include <string.h>
#include <stdint.h>
#include "gim_types.h"

 TNode* newTNode() {
	TNode *node = malloc(sizeof(TNode));
	node ->Parent = NULL;
	node -> Children = NULL;
	node -> data = NULL;
	node -> type = NULL;
	node ->path = NULL;
	return node;
}


GimMasterTree* NewGimMasterTree(){
	GimMasterTree *mastertree = malloc(sizeof(GimMasterTree));
	mastertree ->  MasterDir = NULL;
	mastertree -> MasterRoot = NULL;
	mastertree -> MasterRecordDir = NULL;
	return mastertree;
}
